google/cloud/sql/connector/connector.py

```diff
@@ -321,6 +323,14 @@ async def connect_async(
         # attempt to make connection to Cloud SQL instance
         try:
             conn_info = await cache.connect_info()
+            # validate driver matches intended database engine
+            mapping = DriverMapping[driver.upper()]
```

Comment:
What if we encapsulated more details and did something like this?

``` python
conn_info = await cache.connect_info()
if not DriverMapping.matching_engine(driver, conn_info.database_version):
    raise IncompatibleDriverError
```

where the DriverMapping method looks like:

``` python
class DriverMapping(Enum):
    def matching_engine(driver: str, engine: str) -> bool:
        return engine.startswith(DriverMapping[driver.upper()])
```

Reply:
this would mean i no longer have access to the proper compatible database engine `mapping.value` in the error message without doing a separate call to `DriverMapping[driver.upper()].value` which is more work...

Reply:
Just an idea -- feel free to handle it however you like. A good error message is most important.